### DISCUSSION TITLE
A way to get MBB protocol

### DISCUSSION THREAD

**Discussion by yuanqiuxing** [TheLastGimbus/FreeBuddy#12] [Q&A]
Hi, Dear Author:

Could you please tell me how you get the MBB protocol?

Thanks

**Comment by TheLastGimbus** [comment]
Hi!

I first tried to snoop the Ai Life's activity with wireshark + adb bluetooth debugging, but it turned out that app logs all bluetooth communication in logcat, so that's easier

Other guy who made https://github.com/melianmiko/OpenFreebuds/ also made a great blog about the protocol, check it out: https://mmk.pw/en/posts/freebuds-4i-proto/

Here is also my wiki, tho it's bit harder to read honestly: https://github.com/TheLastGimbus/FreeBuddy/blob/master/notes/mbb-protocol-wiki.md

(in general, i have no idea if it's actually named MBB - found 0 references to it online, but that's what pops up while rev-enging the official app :eyes:)

Ps in general, when i will involve more people to freebuddy, I plan to make some framework for rev-enigng any buds in general - but for now, if you'd like to help/need help for other project, feel free to chat on Discord :+1:
